migrationTool/ams/AssetMigrationTracker.cs

```diff
@@ -129,9 +129,9 @@ public async Task UpdateMigrationStatus(BlobContainerClient container, AssetMigr
                 metadata.Add(ManifestNameKey, result.ManifestName);
             }
 
-            if (result.OutputPath != null)
+            if (result.OutputPath != null && !string.IsNullOrEmpty(result.OutputPath.AbsoluteUri))
             {
-                metadata.Add(OutputPathKey, result.OutputPath.AbsoluteUri);
+                metadata.Add(OutputPathKey, Uri.EscapeDataString(result.OutputPath.AbsoluteUri));
```

Comment:
This doesn't need escaping since Uri.AbsoluteUri is already escaped.  Instead you need escaping on line 129 where the manifest name can contain unicode characters.